### PR TITLE
Fixed double-writes.

### PR DESCRIPTION
**Pull request by AlexLoyko**
### Description
* File is open multiple times and overwritten when generating API modules. Mini-optimization to avoid it.

### Issues Resolved
None, self-research

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,8 +6,10 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Added option to pass custom headers to 'AWSV4SignerAsyncAuth' ([863](https://github.com/opensearch-project/opensearch-py/pull/863))
 - Added sync and async sample that uses `search_after` parameter ([859](https://github.com/opensearch-project/opensearch-py/pull/859))
 ### Updated APIs
+- Updated opensearch-py APIs to reflect [opensearch-api-specification@a2afe56](https://github.com/opensearch-project/opensearch-api-specification/commit/a2afe567254c42aee70b74a8d7d7228c982b56db)
```
Comment (dblock) [inline CHANGELOG.md:9]: This seems unrelated, remove.

File: test_opensearchpy/test_async/test_connection.py [hunk 2]
```diff
@@ -236,7 +236,7 @@ async def test_nowarn_when_test_uses_https_if_verify_certs_is_off(self) -> None:
                 assert (
                     str(w[0].message) == "enable_cleanup_closed ignored because "
                     "https://github.com/python/cpython/pull/118960 is fixed in "
-                    "Python version sys.version_info(major=3, minor=12, micro=7, releaselevel='final', serial=0)"
+                    "Python version sys.version_info(major=3, minor=12, micro=8, releaselevel='final', serial=0)"
```
Comment (dblock) [inline test_opensearchpy/test_async/test_connection.py:239]: This fine, but I'd prefer we fixed it once and for all by checking that the string contains `https://github.com/python/cpython/pull/118960` and none of the other stuff.

**Review by dblock — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (AlexLoyko) [reply CHANGELOG.md:9]: was confused because it auto-popped somehow

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
@AlexLoyko not sure why, but CI on main failed (and passed here)

Can you please check https://github.com/opensearch-project/opensearch-py/actions/runs/12437036439/job/34726093595? PR fixes.

**Comment by AlexLoyko** [comment]
@dblock Not sure why it's failed suddenly on main (likely something about a version not being pinned), but here's a PR that fixes it: https://github.com/opensearch-project/opensearch-py/pull/875. Also, that error was reproduceable on main pre my commit by running `nox -rs generate` or `nox -rs lint` (also hadn't appeared before).
